lib/decompress/zstd_decompress.c

```diff
@@ -936,13 +941,16 @@ static seq_t ZSTD_decodeSequence(seqState_t* seqState, const ZSTD_longOffset_e l
 
     seq.matchLength = ML_base[mlCode]
                     + ((mlCode>31) ? BIT_readBitsFast(&seqState->DStream, mlBits) : 0);  /* <=  16 bits */
-    if (MEM_32bits() && (mlBits+llBits>24)) BIT_reloadDStream(&seqState->DStream);
+    if (MEM_32bits() && (mlBits+llBits>(STREAM_ACCUMULATOR_MIN_32-kMaxExtraBits32)))
+        BIT_reloadDStream(&seqState->DStream);
+    if (MEM_64bits() && (totalBits > STREAM_ACCUMULATOR_MIN_64 - (LLFSELog+MLFSELog+OffFSELog)))
```

Comment:
I now realise that we have a single value `STREAM_ACCUMULATOR_MIN_64` (and other equivalents) applied to both read and write side.
But it's slightly misleading.

On the write side, it's not possible to write more than 56 bits, otherwise the following shift operation may fail.
No such problem on the read side, which means  it's possible to read up to 57 bits.
Not a large difference, but still.

`STREAM_ACCUMULATOR_MIN_64` is set to the larger one, 57 bits.
It's possible to make up for it by using `<` for write side, and `<=` for read side.
But it risks being confusing.

Other solutions : 
- have 2 different macro/variables, depending on read/write side
- settle for the smaller value (56 bits for above example)

What do you think ?

Reply:
If there isn't a penalty for using the smaller value (with the current value of MaxOff), I think we should use it, and maybe note that it could be one larger for read so we don't forget. Otherwise, I think we should have two different macros, since it is too easy to miss that it is different.

Reply:
Agreed.
Let's go for the smaller value.

Let's also check where it's used.
It's quite possible that current tests take care of this difference alternating `<` and `<=` when needed.